src/rules/ec2/EC2EBSVolumeEncrypted.ts

```diff
@@ -91,7 +85,7 @@ function isMatchingLaunchTemplate(
 
 function isLaunchTemplateCompliant(
   node: CfnInstance | CfnAutoScalingGroup
-): boolean {
+): boolean | undefined {
   const launchTemplate = Stack.of(node).resolve(node.launchTemplate);
```

Comment:
Let's move this check outside of this function. I think In this case It makes more sense to check whether the EC2 instance/ ASG has a Launch Template defined before checking if it's compliant. That way we don't have to return `undefined` in this scenario

Reply:
Fixed!